### PR TITLE
Explain how to navigate prestashop

### LINKED ISSUE #939: There is no page that explains how to navigate between URL, controller and view
There is no page that explains how to navigate between URL, controller and view for Front Office and Back Office.

It can be difficult for newcomers to understand by themselves.

### PR DESCRIPTION
**Pull request by matks**
| Questions     | Answers
| ------------- | -------------------------------------------------------
| Branch?       | 8.x
| Description?  | Add a "A Small Tour" page in Introduction page. It tries to explain how to navigate between URL, controller and view for Front Office and Back Office.
| Fixed ticket? | Fixes https://github.com/PrestaShop/docs/issues/939

### CODE REVIEW

File: basics/introduction.md [hunk 1]
```diff
@@ -95,3 +95,60 @@ The driving idea is that we want our code to be more robust, more modular, and f
 Using a proven and popular open-source framework will allow us to focus on our core business code (managing a cart, handling orders, calculating prices and taxes, generating invoices, etc.) with greater efficiency, while enjoying the stability of a globally recognized framework.
 
 In the documentation, we will refer to the 1.6 framework as the "legacy" framework, as this is a popular [designation](https://en.wikipedia.org/wiki/Legacy_system) used in the software world.
+
+
+## How it is built inside
+
+Here is a small guide to help you navigate in the software.
```
Comment (kpodemski) [inline basics/introduction.md]:
```suggestion
Here is a small guide to help you navigate the software.
```

File: basics/introduction.md [hunk 2]
```diff
@@ -95,3 +95,60 @@ The driving idea is that we want our code to be more robust, more modular, and f
...
+
+### Front office MVC
+
+The Front Office is a small application being powered by PHP and Smarty.
```
Comment (kpodemski) [inline basics/introduction.md]:
```suggestion
The Front Office is an application being powered by PHP and Smarty.
```

File: basics/introduction.md [hunk 3]
```diff
@@ -95,3 +95,60 @@ The driving idea is that we want our code to be more robust, more modular, and f
...
+The backend relies on controllers you can find in directory `controllers/front`
+and PrestaShop classes that contain the business logic, mainly from the `classes` folder.
+
+The views and the Javascript come from the installed theme that you will find in `themes/`. You can have multiple themes available on a shop but only one is enabled and in use.
+
+If the Multistore feature is enabled, each shop can use a different theme.
+
+### Visit a Front controller
+
+When you visit `/` on your shop you are being returned the Homepage.
```
Comment (kpodemski) [inline basics/introduction.md]:
```suggestion
When you visit `/` on your shop, you are being returned to the Homepage.
```

File: basics/introduction.md [hunk 4]
```diff
@@ -95,3 +95,60 @@ The driving idea is that we want our code to be more robust, more modular, and f
...
+
+The HTTP request you sent was received by the _Dispatcher_ which found the right Front controller (the `IndexController` for Homepage).
+
+This `IndexController` returned an HTTP response containing an HTML document rendered by Smarty.
+
+## Back office MVC
+
+The Back Office is a large application being powered by PHP and Smarty for the legacy part and by PHP, Symfony and Twig for the migrated part.
```
Comment (kpodemski) [inline basics/introduction.md]:
```suggestion
The Back Office is a large application powered by PHP and Smarty for the legacy part and by PHP, Symfony, and Twig for the migrated part.
```

File: basics/introduction.md [hunk 5]
```diff
@@ -95,3 +95,60 @@ The driving idea is that we want our code to be more robust, more modular, and f
...
+
+The legacy backend relies on controllers which you can find in directory `controllers/admin`. It also relies on PrestaShop classes that contain the business logic, mainly from the `classes` folder.
```
Comment (kpodemski) [inline basics/introduction.md]:
```suggestion
The legacy backend relies on controllers which you can find in the directory `controllers/admin`. It also relies on PrestaShop classes that contain the business logic, mainly from the `classes` folder.
```

File: basics/introduction.md [hunk 6]
```diff
@@ -95,3 +95,60 @@ The driving idea is that we want our code to be more robust, more modular, and f
...
+
+For legacy pages, the views and the Javascript can be found in `admin-dev/themes/default`.
+
+The Symfony backend relies on the controller you can find in `templates/bundles/PrestaShopBundle/Controller/Admin` and PrestaShop logic that mostly comes from the `src` folder, and also some `classes` files.
```
Comment (kpodemski) [inline basics/introduction.md]:
```suggestion
The Symfony backend relies on the controllers you can find in `templates/bundles/PrestaShopBundle/Controller/Admin` and PrestaShop logic that mostly comes from the `src` folder, and also some `classes` files.
```

File: basics/introduction.md [hunk 7]
```diff
@@ -95,3 +95,60 @@ The driving idea is that we want our code to be more robust, more modular, and f
...
+
+For migrated pages, the views can be found in `templates/bundles/PrestaShopBundle/Resources/views`.
```
Comment (kpodemski) [inline basics/introduction.md]:
```suggestion
For migrated pages, you can find the views in `templates/bundles/PrestaShopBundle/Resources/views`.
```

**Comment by matks** [comment]
Thank you for the review @kpodemski all feedbacks applied

File: basics/introduction.md [hunk 8]
```diff
@@ -95,3 +95,60 @@ The driving idea is that we want our code to be more robust, more modular, and f
...
+
+## How it is built inside
+
+Here is a small guide to help you navigate the software.
+
+### Front office MVC
+
+The Front Office is an application being powered by PHP and Smarty.
+
+The backend relies on controllers you can find in directory `controllers/front`
```
Comment (atomiix) [inline basics/introduction.md]:
```suggestion
The backend relies on controllers you can find in the directory `controllers/front`
```

File: basics/introduction.md [hunk 9]
```diff
@@ -95,3 +95,60 @@ The driving idea is that we want our code to be more robust, more modular, and f
...
+
+### Visit a legacy Back Office controller
+
+When you visit `/admin{x}/index.php?controller=AdminCarriers&token={y}` on your shop, you are being returned to the Carriers Back Office page.
+
+The HTTP request you sent was received by the Back Office _Dispatcher_ which found the right Admin controller (the `AdminCarriersController`).
+
+This `AdminCarriersController` returned an HTTP response containing an HTML document rendered by Smarty.
+
+The Admin controllers use a generic system to choose what Smarty template use. Generic templates to display forms and listings are available and the controller provides the structure configuration (for example it controls what columns and rows are displayed).
```
Comment (atomiix) [inline basics/introduction.md]:
```suggestion
The Admin controllers use a generic system to choose what Smarty template to use. Generic templates to display forms and listings are available and the controller provides the structure configuration (for example it controls what columns and rows are displayed).
```

**Review by atomiix — APPROVED** [review]

**Comment by matks** [comment]
Thank you for review

**Comment by eternoendless** [comment]
Needs to be backported into 1.7.x right?

**Pull request merged**